src/app/util/af-types.h

```diff
@@ -224,6 +234,11 @@ struct EmberAfDefinedEndpoint
      * Root endpoint id for composed device type.
      */
     chip::EndpointId parentEndpointId = chip::kInvalidEndpointId;
+
+    /**
+     * Endpoint's composition type
+     */
+    EmberAfEndpointCompositionType endpointCompositionType = EMBER_AF_ENDPOINT_COMPOSITION_FLAT;
```

Comment:
I do wonder whether we should move this into `bitmask` instead of having an enum, etc for this boolean value.  And then we could just have methods on this class for `usesFlatComposition()` and `usesTreeComposition()`?

We might not have enough endpoint definitions around to really care that much about the memory, I guess.

Reply:
> I do wonder whether we should move this into bitmask instead of having an enum

I did think this originally but opted for this to make the PR a bit clearer - I'll try what you suggested though as a do like that.